File: android/src/main/kotlin/cn/mofada/factory_push/constant/MethodName.kt
package cn.mofada.factory_push.constant

/**
 * @author fada
 * @email fada@mofada.cn
 * @date 2020/8/10
 * @description 方法名称
 */
enum class MethodName {
    /**
     * 获取android版本
     */
    GetPlatformVersion,

    /**
     * 初始化
     * @param xiaomiKey
     * @param xiaomi
     */
    Setup,

    /**
     * 设置调试模式
     */
    SetDebugMode,

    /**
     * 停止推送
     */
    StopPush,

    /**
     * 设置别名
     */
    SetAlias,

    /**
     * 取消别名
     */
    DeleteAlias,

    /**
     * 获取所有的别名
     * 
     */
    GetAllAlias,

    /**
     * 清除所有别名
     */
    CleanAlias,

    /**
     * 设置标签, 小米的叫主题
     *
     */
    AddTag,

    /**
     * 设置标签, 小米的叫主题
     *
     */
    AddTags,

    /**
     * 删除标签, 小米的叫主题
     */
    DeleteTag,

    /**
     * 获取所有标签, 小米的叫主题
     */
    GetAllTag,

    /**
     * 清除所有的tag, 小米的叫主题
     */
    CleanTag,

    /**
     * 清除通知
     */
    ClearNotification,

    /**
     * 清除所有的通知
     */
    ClearAllNotification,

    /**
     * 暂停通知
     */
    PausePush,

    /**
     * 恢复通知
     */
    ResumePush,

    /**
     * 启用推送
     */
    EnablePush,

    /**
     * 禁用推送
     */
    DisablePush,

    /**
     * 获取客户端的RegistrationId
     */
    GetRegistrationId,

    /**
     * 设置推送时间
     */
    SetPushTime,

    /**
     * 是否打开通知
     */
    IsNotificationEnabled,

    /**
     * 前往通知打开界面
     */
    OpenNotification,
}

File: lib/constant/method_name.dart
import 'package:factory_push/constant/argument_name.dart';

/// @author : fada
/// @email : fada@mofada.cn
/// @date : 2020/8/11 10:11
/// @description : 方法名称常量

class MethodName {
  ///获取android版本
  static const getPlatformVersion = "GetPlatformVersion";

  ///初始化
  ///@param [ArgumentName.xiaomiAppId]
  ///@param [ArgumentName.xiaomiAppKey]
  static const setup = "Setup";

  ///设置调试模式
  ///@param [ArgumentName.debugMode]
  static const setDebugMode = "SetDebugMode";

  ///注销推送
  static const stopPush = "StopPush";

  ///设置别名
  ///@param [ArgumentName.alias]
  static const setAlias = "SetAlias";

  ///删除别名
  ///@param [ArgumentName.alias]
  static const deleteAlias = "DeleteAlias";

  ///获取所有的别名
  static const getAllAlias = "GetAllAlias";

  ///清除所有别名
  static const cleanAlias = "CleanAlias";

  ///设置标签
  ///@param [ArgumentName.tag]
  static const addTag = "AddTag";

  ///批量设置标签
  ///@param [ArgumentName.tags]
  static const addTags = "AddTags";

  ///删除标签
  ///@param [ArgumentName.tag]
  static const deleteTag = "DeleteTag";

  ///获取所有的标签
  static const getAllTag = "GetAllTag";

  ///清除所有的标签
  static const cleanTag = "CleanTag";

  ///清除指定id的通知
  ///@param [ArgumentName.notify_id]
  static const clearNotification = "ClearNotification";

  ///清除所有的通知
  static const clearAllNotification = "ClearAllNotification";

  ///暂停通知
  static const pausePush = "PausePush";

  ///恢复通知
  static const resumePush = "ResumePush";

  ///启用推送
  static const enablePush = "EnablePush";

  ///禁用推送
  static const disablePush = "DisablePush";

  ///获取注册id
  static const getRegistrationId = "GetRegistrationId";

  ///设置推送时间
  ///@param [ArgumentName.startHour]
  ///@param [ArgumentName.startMinter]
  ///@param [ArgumentName.endHour]
  ///@param [ArgumentName.endMinter]
  static const setPushTime = "SetPushTime";

  ///是否开启了通知
  static const isNotificationEnabled = "IsNotificationEnabled";

  ///打开通知设置界面
  static const openNotification = "OpenNotification";
}
